rig-core/src/providers/cohere.rs

```diff
@@ -63,7 +63,23 @@ impl Client {
     }
 
     pub fn embedding_model(&self, model: &str, input_type: &str) -> EmbeddingModel {
-        EmbeddingModel::new(self.clone(), model, input_type)
+        let ndims = match model {
+            EMBED_ENGLISH_V3 | EMBED_MULTILINGUAL_V3 | EMBED_ENGLISH_LIGHT_V2 => 1024,
+            EMBED_ENGLISH_LIGHT_V3 | EMBED_MULTILINGUAL_LIGHT_V3 => 384,
+            EMBED_ENGLISH_V2 => 4096,
+            EMBED_MULTILINGUAL_V2 => 768,
+            _ => 0,
+        };
```

Comment:
Add dosctrings that mention a default ndims will be used if the model doesnt match one of the enum values